ambry-server/src/main/java/com/github/ambry/server/AmbryRequests.java

```diff
@@ -693,6 +695,34 @@ public void handleReplicaMetadataRequest(NetworkRequest request) throws IOExcept
             logger.trace("{} Time used to find entry since: {}", partitionId,
                 (SystemTime.getInstance().milliseconds() - partitionStartTimeInMs));
 
+            // Compute CRC for backup recovery and verification
+            if (replicaMetadataRequestInfo.getReplicaPath().startsWith(BackupCheckerThread.DR_Verifier_Keyword)) {
+              Store storeToGet = storeManager.getStore(partitionId);
+              EnumSet<StoreGetOptions> storeGetOptions = EnumSet.of(StoreGetOptions.Store_Include_Deleted,
+                  StoreGetOptions.Store_Include_Expired);
+              List<MessageInfo> newMessageInfos = new ArrayList<>();
+              MessageFormatSend msgFmt = new MessageFormatSend();
+              // for-each blob
+              findInfo.getMessageEntries().forEach(minfo -> {
+                MessageInfo newMsgInfo = minfo;
+                try {
+                  List<StoreKey> keys = getConvertedStoreKeys(Collections.singletonList(minfo.getStoreKey()))
+                      .stream().distinct().collect(Collectors.toList());
+                  MessageReadSet rdset = storeToGet.get(keys, storeGetOptions).getMessageReadSet();
```

Comment:
we should only do this when the MessageInfo shows that it's not DELETE, TTL_UPDATE and UNDELETE, that this is a PUT message info.

Reply:
Hmm.. the `Store` returns a single `MessageInfo` with all state changes for a blob including Put, Del, TtlUpdate etc. And the verification app is essentially always bootstrapping as it starts from an empty disk. So the verification app sees just 1 message for a blob with all state changes. And this is its only chance at comparing CRCs. If we exclude messages that have Del, TtlUpdate etc. flags set, then we will be unable to compare CRCs.

Reply:
we would be fine. We probably should remove the delete at least. I think ttl update and undelete should be included. The deleted MessageInfo could indicate that there is no blob content anymore.